### PR TITLE
Merge 'features/compiler' into stackalloc-init

### PR DESCRIPTION
**Pull request by alrz**

### CODE REVIEW

**Comment by alrz** [comment]
resolved conflicts:


```
	src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs
	src/Compilers/CSharp/Portable/BoundTree/Formatting.cs
	src/Compilers/CSharp/Portable/CSharpResources.resx
	src/Compilers/CSharp/Portable/Errors/MessageID.cs
	src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt
	src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs
	src/Compilers/Core/Portable/CodeGen/ILBuilderEmit.cs
```
plus a follow up commit added.

File: src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs [hunk 1]
```diff
@@ -350,7 +350,7 @@ void Method1(S s, bool c)
     {
         var obj1 = c ? default : stackalloc[] { """" };
         var obj2 = c ? default : stackalloc[] { new {} };
-        var obj3 = c ? default : stackalloc[] { s }; // Should be an error, see https://github.com/dotnet/roslyn/issues/25086
+        var obj3 = c ? default : stackalloc[] { s };
```
Comment (VSadov) [inline src/Compilers/CSharp/Test/Semantic/Semantics/StackAllocInitializerTests.cs]: nice!!

**Comment by VSadov** [comment]
LGTM
there seems to be conflicts with the recent change.

**Comment by alrz** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by VSadov** [comment]
@dotnet-bot test ubuntu_16_mono_debug_prtest  please

**Comment by VSadov** [comment]
`SIGABRT while executing native code`  on Mono.
I think we can ignore that.

**Pull request merged**